RT_SYCL/material.hpp

```diff
@@ -78,7 +78,7 @@ struct dielectric_material {
         attenuation *= albedo;
         float refraction_ratio = rec.front_face ? (1.0f / ref_idx) : ref_idx;
         vec unit_direction = unit_vector(r_in.direction());
-        float cos_theta = sycl::fmin(sycl::dot(-unit_direction, rec.normal), 1.0f);
+        float cos_theta = sycl::fmin(sycl::dot(vec{}-unit_direction, rec.normal), 1.0f);
```

Comment:
This is more efficient anyway:
```suggestion
        float cos_theta = sycl::fmin(-sycl::dot(unit_direction, rec.normal), 1.0f);
```